CUDA: tests fail with `TypeError: can_cast()`

### Version of Awkward Array

2.6.7

### Description and code to reproduce

25 tests-cuda and 131 tests-cuda-kernels-explicit fail what looks like in min/max reducers with `TypeError: can_cast()` 
```
awkward-cpp 37
cupy 13.2.0
cuda-version 12.6

```
```python
tests-cuda-kernels-explicit/test_unit_cudaawkward_reduce_min_uint8_uint8_64.py:47:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
src/awkward/_kernels.py:169: in __call__
    self._impl(grid, blocks, args)
        ak_cuda    = <module 'awkward._connect.cuda' from '/home/yana/Projects/PR3205/awkward/src/awkward/_connect/cuda/__init__.py'>
        args       = (array([123, 123, 123, 123], dtype=uint8), array([1, 3, 5, 4, 2, 2, 3, 1, 5], dtype=uint8), array([0, 0, 0, 0, 0, 2, 2, 2, 3]), 9, 4, 4, ...)
        blocks     = (9, 1, 1)
        cupy       = <module 'cupy' from '/home/yana/miniconda3/envs/awkward-cuda/lib/python3.12/site-packages/cupy/__init__.py'>
        cupy_stream_ptr = 0
        grid       = (1, 1, 1)
        maxlength  = 9
        self       = <CupyKernel awkward_reduce_min, uint8, uint8, int64>
src/awkward/_connect/cuda/_kernel_signatures.py:2969: in f
    temp = cupy.full(lenparents, identity, dtype=toptr.dtype)
        args       = (array([123, 123, 123, 123], dtype=uint8), array([1, 3, 5, 4, 2, 2, 3, 1, 5], dtype=uint8), array([0, 0, 0, 0, 0, 2, 2, 2, 3]), 9, 4, 4, ...)
        block      = (9, 1, 1)
        cuda_kernel_templates = <cupy._core.raw.RawModule object at 0x7f233daaeda0>
        err_code   = array(18446744073709551615, dtype=uint64)
        fromptr    = array([1, 3, 5, 4, 2, 2, 3, 1, 5], dtype=uint8)
        grid       = (1, 1, 1)
        grid_size  = 1
        identity   = 4
        invocation_index = 411
        lenparents = 9
        outlength  = 4
        parents    = array([0, 0, 0, 0, 0, 2, 2, 2, 3])
        toptr      = array([123, 123, 123, 123], dtype=uint8)
../../../miniconda3/envs/awkward-cuda/lib/python3.12/site-packages/cupy/_creation/basic.py:325: in full
    cupy.copyto(a, fill_value, casting='unsafe')
        a          = array([123,   0,   0,   0,   0,   0,   0,   0, 123], dtype=uint8)
        dtype      = dtype('uint8')
        fill_value = 4
        order      = 'C'
        shape      = 9
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

dst = array([123,   0,   0,   0,   0,   0,   0,   0, 123], dtype=uint8), src = 4, casting = 'unsafe', where = None

    def copyto(dst, src, casting='same_kind', where=None):
        """Copies values from one array to another with broadcasting.

        This function can be called for arrays on different devices. In this case,
        casting, ``where``, and broadcasting is not supported, and an exception is
        raised if these are used.

        Args:
            dst (cupy.ndarray): Target array.
            src (cupy.ndarray): Source array.
            casting (str): Casting rule. See :func:`numpy.can_cast` for detail.
            where (cupy.ndarray of bool): If specified, this array acts as a mask,
                and an element is copied only if the corresponding element of
                ``where`` is True.

        .. seealso:: :func:`numpy.copyto`

        """
        src_is_numpy_scalar = False

        src_type = type(src)
        src_is_python_scalar = src_type in (
            int, bool, float, complex,
            fusion._FusionVarScalar, _fusion_interface._ScalarProxy)
        if src_is_python_scalar:
            src_dtype = numpy.dtype(type(src))
>           can_cast = numpy.can_cast(src, dst.dtype, casting)
E           TypeError: can_cast() does not support Python ints, floats, and complex because the result used to depend on the value.
E           This change was part of adopting NEP 50, we may explicitly allow them again in the future.

casting    = 'unsafe'
dst        = array([123,   0,   0,   0,   0,   0,   0,   0, 123], dtype=uint8)
src        = 4
src_dtype  = dtype('int64')
src_is_numpy_scalar = False
src_is_python_scalar = True
src_type   = <class 'int'>
where      = None

../../../miniconda3/envs/awkward-cuda/lib/python3.12/site-packages/cupy/_manipulation/basic.py:38: TypeError

```

--- a/src/awkward/_broadcasting.py
+++ b/src/awkward/_broadcasting.py
@@ -712,7 +712,7 @@ def apply_step(
                     mask = backend.index_nplike.logical_or(mask, m, maybe_out=mask)
 
         nextmask = Index8(mask.view(np.int8))
-        index = backend.index_nplike.full(mask.shape[0], -1, dtype=np.int64)
+        index = backend.index_nplike.full(mask.shape[0], np.int64(-1), dtype=np.int64)
         index[~mask] = backend.index_nplike.arange(
             backend.index_nplike.shape_item_as_index(mask.shape[0])
             - backend.index_nplike.count_nonzero(mask),

--- a/src/awkward/_connect/cuda/cuda_kernels/awkward_NumpyArray_subrange_equal_bool.cu
+++ b/src/awkward/_connect/cuda/cuda_kernels/awkward_NumpyArray_subrange_equal_bool.cu
@@ -4,9 +4,9 @@
 // def f(grid, block, args):
 //     (tmpptr, fromstarts, fromstops, length, toequal, invocation_index, err_code) = args
 //     if length > 1:
-//         scan_in_array = cupy.full((length - 1) * (length - 2), 0, dtype=cupy.int64)
+//         scan_in_array = cupy.full((length - 1) * (length - 2), cupy.array(0), dtype=cupy.int64)
 //     else:
-//         scan_in_array = cupy.full(0, 0, dtype=cupy.int64)
+//         scan_in_array = cupy.full(0, cupy.array(0), dtype=cupy.int64)
 //     cuda_kernel_templates.get_function(fetch_specialization(["awkward_NumpyArray_subrange_equal_bool", bool_, fromstarts.dtype, fromstops.dtype, bool_]))(grid, block, (tmpptr, fromstarts, fromstops, length, toequal, scan_in_array, invocation_index, err_code))
 //     toequal[0] = cupy.any(scan_in_array == True)
 // out["awkward_NumpyArray_subrange_equal_bool", {dtype_specializations}] = None

--- a/src/awkward/_connect/cuda/cuda_kernels/awkward_reduce_max.cu
+++ b/src/awkward/_connect/cuda/cuda_kernels/awkward_reduce_max.cu
@@ -7,7 +7,7 @@
 //         grid_size = math.floor((lenparents + block[0] - 1) / block[0])
 //     else:
 //         grid_size = 1
-//     temp = cupy.full(lenparents, identity, dtype=toptr.dtype)
+//     temp = cupy.full(lenparents, cupy.array([identity]), dtype=toptr.dtype)
 //     cuda_kernel_templates.get_function(fetch_specialization(["awkward_reduce_max_a", cupy.dtype(toptr.dtype).type, cupy.dtype(fromptr.dtype).type, parents.dtype]))((grid_size,), block, (toptr, fromptr, parents, lenparents, outlength, toptr.dtype.type(identity), temp, invocation_index, err_code))
 //     cuda_kernel_templates.get_function(fetch_specialization(["awkward_reduce_max_b", cupy.dtype(toptr.dtype).type, cupy.dtype(fromptr.dtype).type, parents.dtype]))((grid_size,), block, (toptr, fromptr, parents, lenparents, outlength, toptr.dtype.type(identity), temp, invocation_index, err_code))
 // out["awkward_reduce_max_a", {dtype_specializations}] = None

--- a/src/awkward/_connect/cuda/cuda_kernels/awkward_reduce_min.cu
+++ b/src/awkward/_connect/cuda/cuda_kernels/awkward_reduce_min.cu
@@ -7,7 +7,8 @@
 //         grid_size = math.floor((lenparents + block[0] - 1) / block[0])
 //     else:
 //         grid_size = 1
-//     temp = cupy.full(lenparents, identity, dtype=toptr.dtype)
+//     dtype=toptr.dtype
+//     temp = cupy.full(lenparents, cupy.array([identity]), dtype=toptr.dtype)
 //     cuda_kernel_templates.get_function(fetch_specialization(["awkward_reduce_min_a", cupy.dtype(toptr.dtype).type, cupy.dtype(fromptr.dtype).type, parents.dtype]))((grid_size,), block, (toptr, fromptr, parents, lenparents, outlength, toptr.dtype.type(identity), temp, invocation_index, err_code))
 //     cuda_kernel_templates.get_function(fetch_specialization(["awkward_reduce_min_b", cupy.dtype(toptr.dtype).type, cupy.dtype(fromptr.dtype).type, parents.dtype]))((grid_size,), block, (toptr, fromptr, parents, lenparents, outlength, toptr.dtype.type(identity), temp, invocation_index, err_code))
 // out["awkward_reduce_min_a", {dtype_specializations}] = None

--- a/src/awkward/_connect/cuda/cuda_kernels/awkward_reduce_min.cu
+++ b/src/awkward/_connect/cuda/cuda_kernels/awkward_reduce_min.cu
@@ -7,7 +7,6 @@
 //         grid_size = math.floor((lenparents + block[0] - 1) / block[0])
 //     else:
 //         grid_size = 1
-//     dtype=toptr.dtype
 //     temp = cupy.full(lenparents, cupy.array([identity]), dtype=toptr.dtype)
 //     cuda_kernel_templates.get_function(fetch_specialization(["awkward_reduce_min_a", cupy.dtype(toptr.dtype).type, cupy.dtype(fromptr.dtype).type, parents.dtype]))((grid_size,), block, (toptr, fromptr, parents, lenparents, outlength, toptr.dtype.type(identity), temp, invocation_index, err_code))
 //     cuda_kernel_templates.get_function(fetch_specialization(["awkward_reduce_min_b", cupy.dtype(toptr.dtype).type, cupy.dtype(fromptr.dtype).type, parents.dtype]))((grid_size,), block, (toptr, fromptr, parents, lenparents, outlength, toptr.dtype.type(identity), temp, invocation_index, err_code))

--- a/src/awkward/_nplikes/array_module.py
+++ b/src/awkward/_nplikes/array_module.py
@@ -118,7 +118,7 @@ class ArrayModuleNumpyLike(NumpyLike[ArrayLikeT]):
         *,
         dtype: DTypeLike | None = None,
     ) -> ArrayLikeT:
-        return self._module.full(shape, fill_value, dtype=dtype)
+        return self._module.full(shape, self._module.array(fill_value), dtype=dtype)
 
     def zeros_like(
         self, x: ArrayLikeT | PlaceholderArray, *, dtype: DTypeLike | None = None
@@ -146,7 +146,9 @@ class ArrayModuleNumpyLike(NumpyLike[ArrayLikeT]):
         if isinstance(x, PlaceholderArray):
             return self.full(x.shape, fill_value, dtype=dtype or x.dtype)
         else:
-            return self._module.full_like(x, fill_value, dtype=dtype)
+            return self._module.full_like(
+                x, self._module.array(fill_value), dtype=dtype
+            )
 
     def arange(
         self,

